### ISSUE TITLE
Errors when Godot try to import files which contains in name unallowed characters

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#27081]
**Godot version:**
Godot 3.1

**OS/device including version:**
Ubuntu 18.04

**Issue description:**
When I put into godot project, folders with names like :::\\::: (kopia) or ::::\\::Zdj?>, then Godot show a lot of errors:
```
ERROR: _scan_filesystem: Condition ' split.size() != 3 ' is true. Continuing..:
   At: editor/editor_file_system.cpp:237.
ERROR: _get_modified_time: Failed to get modified time for: res://::::\\::Zdj (kopia)
   At: drivers/unix/file_access_unix.cpp:292.
ERROR: _scan_new_dir: Cannot go into subdir: new folder
   At: editor/editor_file_system.cpp:736.
```

I think that godot should change name all of names which contains unallowed characters to corrent, like "::::??Zdjecia" -> "Zdjecia"

**Minimal reproduction project:**
[FFFFFFFFFFFFFFFFFFFF.zip](https://github.com/godotengine/godot/files/2969941/FFFFFFFFFFFFFFFFFFFF.zip)

**Comment by akien-mga** [comment]
> I think that godot should change name all of names which contains unallowed characters to corrent, like "::::??Zdjecia" -> "Zdjecia"

I disagree, Godot should not mess around with your filesystem unless you ask it to. It should just refuse to open these folders/import these files and throw a clearer error.
